boards/microbit_v2/src/main.rs

```diff
@@ -522,6 +530,19 @@ pub unsafe fn main() {
         512
     ));
 
+    //Nonvolatile_storage
+    let nonvolatile_storage = components::nonvolatile_storage::NonvolatileStorageComponent::new(
```

Comment:
I think it is worth pulling the relevant parts of the `nonvolatile_storage` driver into a new capsule that can be customized for this use case. In particular, I think we want to avoid giving the OTA app blanket access to all of process flash. Instead, the userspace code should notify the kernel it wants to load a new app, tell the kernel the size of the new app, and then if the kernel is ok with it, it can use the revamped nonvolatile storage to actually write the app binary to flash. However, the OTA app should be able to write the app starting at address 0, and the capsule handles the offset to the actual address in flash. That way the OTA app doesn't need to keep track of the actual flash addresses and memory mapping.

Reply:
That's a good point! I will try to do that.